src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/tables/Client.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.training.designpatterns.fastfoodrestaurant.tables;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.foods.Food;
+import com.epam.training.designpatterns.fastfoodrestaurant.foods.Menu;
+import com.epam.training.designpatterns.fastfoodrestaurant.staff.Server;
+
+import java.util.Random;
+import java.util.concurrent.Semaphore;
+
+/**
+ * Knows the server, and the menu
+ */
+public class Client implements Runnable{
+    private static int clientCount;
+    private final Semaphore sema;
+
+    private double happiness = 100;
+    private final int clientNumber;
+    private Food food;
+    private final int SIMULATION_SPEED;
+    private final Server server;
+
+    public Client(Server server, Semaphore sema, int simulation_speed){
+        SIMULATION_SPEED = simulation_speed;
+        this.clientNumber = clientCount++;
+        this.sema = sema;
+        this.server = server;
+    }
+    private void orderFood() throws InterruptedException {
+        Food food = Menu.randomFoodAndCondiment();
+        System.out.println(String.format("%s ordering %s. Current happiness: %.2f",
+                this.toString(), food.toString(), happiness));
+        server.takeOrder(this, food);
+    }
+
+    public void giveFood(Food food) throws InterruptedException {
```

Comment:
Ha a `waitForFood()` synchronized, akkor ennek is annak kell lennie, nem?

Reply:
A `waitForFood()` azért synchronized, hogy a vendég tudjon várni, a `wait()` miatt. A Server-en a `serveFood()` -on, pedig, hogy felkeltse a várakozó threadet. Magára a cliensre van synchronizálva, ami nem a legjobb megoldás, csak ki akartam ezt is próbálni :)